src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.iOS.cs

```diff
@@ -1,58 +1,79 @@
 ﻿#if XAMARIN_IOS
+
 using System;
 using System.Collections.Generic;
 using System.Text;
+using Uno.UI;
 using Uno.UI.Common;
+using Uno.UI.Helpers.WinUI;
+using Windows.UI.Xaml.Automation;
 using Windows.UI.Xaml.Controls.Primitives;
 using Windows.UI.Xaml.Data;
 
 namespace Windows.UI.Xaml.Controls
 {
-	//TODO: should inherit from PickerFlyoutBase
 	public partial class TimePickerFlyout : Flyout
 	{
-		private bool _hasAcceptButton = false;
+        private bool AlreadyCancelledOrSaved { get; set; }
 
 		public TimePickerFlyout()
 		{
 			InitializeContent();
 		}
 
-		private void InitializeContent()
+		private void onTap(object sender, Input.PointerRoutedEventArgs e) => e.Handled = true;
+
+		protected internal override void Close()
 		{
-			Content = new TimePickerSelector();
-			BindToContent(nameof(Time));
-			BindToContent(nameof(ClockIdentifier));
+			var timePicker = _presenter as TimePickerFlyoutPresenter;
+			var headerTapZone = timePicker.FindName("HeaderTapZone") as Border;
+			headerTapZone.PointerPressed -= onTap;
 
-			AttachAcceptCommand((TimePickerSelector)Content);
+            if (AlreadyCancelledOrSaved == false)
```

Comment:
Why do we need this AlreadyCancelledOrSaved check?

Reply:
since the flyout is not created everytime and stay in the background
tap outside of flyout (flyoutscrim) also need to call cancel to revert the time of the timepicker

so this was a little hack to make sure cancel was always being call on popup close
 
i did this code when i was use the background brush of the flyoutpopuppanel.. i didnt have the flyout scrim at that time as a layer i could pick up.. 

now i refactored that code to use tap event on the flyout scrim but the more i think about it the more i prefer the alreadysaveorcancelled option since it mean at every popup close whatever how we make sure if it wasnt alreadysavedorcancel we cancel and revert the nativepicker to original state..  

what do you prefer ?? this is the other way ... on popup open and unregistring on popup close.. 

var flyoutScrim = timePickerPresenter.FindName("FlyoutScrim") as FrameworkElement;

if (flyoutScrim != null)
{
	flyoutScrim.PointerPressed += OnFlyoutScrimTap;
}

private void OnFlyoutScrimTap(object sender, Input.PointerRoutedEventArgs e) => Dismiss();

Reply:
removed alreadysaved like mat wanted.. now calling cancel on popup closed even after saved ..